cpp/src/barretenberg/dsl/acir_proofs/acir_proofs.cpp

```diff
@@ -154,4 +189,194 @@ bool verify_proof(
     return verified;
 }
 
+/**
+ * @brief Enables simulation of recursion in the ACVM. In order to be able to simulate execute of ACIR,
+ *        a DSL needs to be able to insert into a circuit's witness the correct output aggregation state of a recursive
+ * verification. The resulting aggergation state is what will be compared against in the recursion constraint.
+ *
+ * @param proof_buf - The proof to be verified
+ * @param proof_length - The length of the proof to enable reading the proof from buffer
+ * @param vk_buf - The verification key of the circuit being verified
+ * @param vk_length - The length of the verification eky to enable reading the proof from buffer
+ * @param public_inputs_buf - The public inputs of the proof being verified. To be included when construct the input
+ * aggregation state
+ * @param input_aggregation_obj_buf - The fields represnting the two G1 points that must be fed into a pairing.
+ *                                    This will be empty if this is the first proof to be recursively.
+ * @param output_aggregation_obj_buf - Two G1 points that the top-level verifier needs to run a pairing upon to complete
+ * verification
+ */
+size_t verify_recursive_proof(uint8_t const* proof_buf,
+                              uint32_t proof_length,
+                              uint8_t const* vk_buf,
+                              uint32_t vk_length,
+                              uint32_t num_public_inputs,
+                              uint8_t const* input_aggregation_obj_buf,
+                              uint8_t** output_aggregation_obj_buf)
+{
+
+    bool inner_aggregation_all_zero = true;
+    std::vector<barretenberg::fr> aggregation_input(16);
```

Comment:
We should document what this magic constant 16 is

Reply:
I replaced its usage with `acir_format::RecursionConstraint::AGGREGATION_OUTPUT_STATE`